other/perun-propagate/etc/perun_propagate

```diff
@@ -55,12 +65,26 @@ perun_propagate_start() {
 	fi
 
 	set +e
-	if which timeout stdbuf &>/dev/null; then
-		OUT=`timeout ${TIMEOUT} stdbuf -oL -eL remctl "${PERUN_SERVER}" perun propagate 2>&1`
-		EXIT_CODE=$?
+	for i in $(seq 1 $TIMEOUT)
+	do
+		is_process_running $PID
+		RUNNING=$?
+		if [ "$RUNNING" -eq 1 ]; then
+			sleep 1
+		else
+			wait %1
+			EXIT_CODE=$?
+			break
+		fi
+	done
+
+	if [ -z "${EXIT_CODE}" ]; then
+		OUT="Timeout elapsed, but script is still running."
+		echo "$OUT" >&2
+		EXIT_CODE=1;
 	else
-		OUT=`remctl "${PERUN_SERVER}" perun propagate 2>&1`
-		EXIT_CODE=$?
+		OUT=`cat "$TEMP_MSG"`
+		rm -f "$TEMP_MSG"
```

Comment:
Don't know what this piece of code is really doing. Also please add restart to the start section.